int64 v_right
int64 v_left
int64 v_back